File: server/bcephal-grid/src/main/java/com/moriset/bcephal/grid/domain/JoinEditorData.java
/**
 * 
 */
package com.moriset.bcephal.grid.domain;

import java.util.ArrayList;
import java.util.List;

import com.moriset.bcephal.domain.EditorData;
import com.moriset.bcephal.domain.Nameable;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Joseph Wambo
 *
 */
@Data
@EqualsAndHashCode(callSuper = false)
public class JoinEditorData extends EditorData<Join> {

	private List<Nameable> routines;
	
	private List<Nameable> sequences;
	
	private List<AbstractSmartGrid<?>> grids;
	
	public JoinEditorData() {
		super();
		routines = new ArrayList<>(0);
		grids = new ArrayList<>(0);
	}
	
	public JoinEditorData(EditorData<Join> data) {
		super(data);
		routines = new ArrayList<>(0);
		sequences = new ArrayList<>(0);
		grids = new ArrayList<>(0);
	}
	
}

File: client/bcephal-client-model/Joins/JoinEditorData.cs
using Bcephal.Models.Base;
using Bcephal.Models.Grids;
using System.Collections.ObjectModel;

namespace Bcephal.Models.Joins
{
    public class JoinEditorData : EditorData<Join>
    {
        public ObservableCollection<Nameable> Routines;

        public ObservableCollection<Nameable> Sequences;

        public ObservableCollection<SmartGrille> Grids;

        public JoinEditorData()
        {
            Routines = new ObservableCollection<Nameable>();
            Sequences = new ObservableCollection<Nameable>();
            Grids = new ObservableCollection<SmartGrille>();
        }

    }
}
